var/spack/repos/builtin/packages/intel/package.py

```diff
@@ -27,95 +27,7 @@
-
-
-def get_all_components():
-    """Returns a list of all the components associated with the downloaded
-       Intel package"""
-    all_components = []
-    with open("pset/mediaconfig.xml", "r") as f:
-        lines = f.readlines()
-        for line in lines:
-            if line.find('<Abbr>') != -1:
-                component = line[line.find('<Abbr>') + 6:line.find('</Abbr>')]
-                all_components.append(component)
-    return all_components
-
-
-class IntelInstaller(Package):
-    """Base package containing common methods for installing Intel software"""
-
-    homepage = "https://software.intel.com/en-us"
-    intel_components = "ALL"
-    license_comment = '#'
-    license_files = ['Licenses/license.lic']
-    license_vars = ['INTEL_LICENSE_FILE']
-    license_url = \
-        'https://software.intel.com/en-us/articles/intel-license-manager-faq'
-
-    @property
-    def license_required(self):
-        # The Intel libraries are provided without requiring a license as of
-        # version 2017.2. Trying to specify the license will fail. See
-        # https://software.intel.com/en-us/articles/free-mkl
-        if (self.spec.satisfies("intel-mkl@2017.2:") or
-            self.spec.satisfies("intel-daal@2017.2:") or
-            self.spec.satisfies("intel-ipp@2017.2:")):
-            return False
-        return True
-
-    @property
-    def global_license_file(self):
-        """Returns the path where a global license file should be stored."""
-        if not self.license_files:
-            return
-        return join_path(self.global_license_dir, "intel",
-                         os.path.basename(self.license_files[0]))
-
-    def install(self, spec, prefix):
-
-        if not hasattr(self, "intel_prefix"):
-            self.intel_prefix = self.prefix
-
-        silent_config_filename = 'silent.cfg'
-        with open(silent_config_filename, 'w') as f:
-            f.write("""
-ACCEPT_EULA=accept
-PSET_MODE=install
-CONTINUE_WITH_INSTALLDIR_OVERWRITE=yes
-PSET_INSTALL_DIR=%s
-NONRPM_DB_DIR=%s
-CONTINUE_WITH_OPTIONAL_ERROR=yes
-COMPONENTS=%s
-""" % (self.intel_prefix, self.intel_prefix, self.intel_components))
-
-        # The Intel libraries are provided without requiring a license as of
-        # version 2017.2. Trying to specify the license will fail. See
-        # https://software.intel.com/en-us/articles/free-mkl
-        if not (spec.satisfies("intel-mkl@2017.2:") or
-                spec.satisfies("intel-daal@2017.2:") or
-                spec.satisfies("intel-ipp@2017.2:")):
-            with open(silent_config_filename, 'a') as f:
-                f.write("""
-ACTIVATION_LICENSE_FILE=%s
-ACTIVATION_TYPE=license_file
-PHONEHOME_SEND_USAGE_DATA=no
-""" % (self.global_license_file))
-
-        install_script = Executable("./install.sh")
-        install_script('--silent', silent_config_filename)
-
-
-class Intel(IntelInstaller):
+class Intel(IntelPackage):
```

Comment:
We do, see #3973 for the reason why. If anything, I would rather get rid of the `intel-parallel-studio` package. But we wouldn't be able to get the version numbers to match up unless we could have different versions for different variants.